app/controllers/institutions_controller.rb

```diff
@@ -152,7 +152,7 @@ def set_locations
         end
         lat += loc.lat
         long += loc.long
-        gon.markers << [i.id, i.name, loc.lat, loc.long]
+        gon.markers << [i.id, i.name, loc.lat, loc.long, ActionController::Base.helpers.link_to(i.name, institution_path(i.id))]
```

Comment:
I would personally switch `i.name` to the new code instead as mentioned previously.